### PR TITLE
Schnorr Signature support

### PR DESCRIPTION
**Pull request by jimmysong**

### CODE REVIEW

**Comment by jimmysong** [comment]
WIP, still working on the error test vectors and `cecc.py`

**Comment by mflaxman** [comment]
One idea to make this easier to merge: make schnorr support python only for now.

**Comment by jimmysong** [comment]
OK, this should work now. Requires libsecp256k1 to be configured with

`$ ./configure --enable-module-extrakeys --enable-module-schnorrsig --enable-experimental`

The libsec optimization is totally worth it. `test_schnorr` on my machine runs in 0.05 seconds with libsec, 5.8 seconds without.

**Comment by mflaxman** [comment]
> OK, this should work now. Requires libsecp256k1 to be configured with
> `$ ./configure --enable-module-extrakeys --enable-module-schnorrsig --enable-experimental`

Add this to readme as part of this PR?

File: buidl/schnorr.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    "nonce": sha256(b"BIP0340/nonce") + sha256(b"BIP0340/nonce"),
+    "challenge": sha256(b"BIP0340/challenge") + sha256(b"BIP0340/challenge"),
+}
+
+
+def tagged_hash(tag, msg):
+    return sha256(TAGS[tag] + msg)
+
+
+def xor_bytes(a, b):
```
Comment (mflaxman) [inline buidl/schnorr.py:15]: Does it make sense for this to be a dedicated helper method? it's only called 1x.

**Review by mflaxman — APPROVED** [review]
Awesome!

Didn't test libsec as I don't think brew on MacOS currently supports the `--enable-module-extrakeys` and `--enable-module-schnorrsig` flags (`--enable-experimental` is enabled by default though):
https://github.com/cuber/homebrew-libsecp256k1/blob/master/libsecp256k1.rb#L16

It looks like you could host your own `libsecp256k1.rb ` though fairly easily.

Bump the version in `setup.py`?

**Comment by jimmysong** [comment]
all bumped and README instructions updated.

**Pull request merged**